src/main/java/com/ita/if103java/ims/dao/impl/UserDaoImpl.java

```diff
@@ -0,0 +1,218 @@
+package com.ita.if103java.ims.dao.impl;
+
+import com.ita.if103java.ims.dao.UserDao;
+import com.ita.if103java.ims.entity.User;
+import com.ita.if103java.ims.exception.CRUDException;
+import com.ita.if103java.ims.exception.EntityNotFoundException;
+import com.ita.if103java.ims.mapper.jdbc.UserRowMapper;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.dao.DataAccessException;
+import org.springframework.dao.EmptyResultDataAccessException;
+import org.springframework.jdbc.core.JdbcTemplate;
+import org.springframework.jdbc.support.GeneratedKeyHolder;
+import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder;
+import org.springframework.stereotype.Repository;
+
+import javax.sql.DataSource;
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.time.ZoneId;
+import java.time.ZonedDateTime;
+import java.util.List;
+import java.util.Optional;
+import java.util.UUID;
+
+@Repository
+public class UserDaoImpl implements UserDao {
+
+    private static UserRowMapper userRowMapper;
+    private static JdbcTemplate jdbcTemplate;
+    private static BCryptPasswordEncoder bCryptPasswordEncoder;
```

Comment:
remove static for these fields.

In previous comment I mean that it is good practice to declare first static fields and then non static in the class, so just move declaration from row 35 up after class declaration and before these fields.